x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java

```diff
@@ -520,9 +522,26 @@ public void resolveAsSeparateMappings(String indexWildcard, String javaRegex, bo
                         }
                         EsField field = indexFields.flattedMapping.get(fieldName);
                         if (field == null || (invalidField != null && (field instanceof InvalidMappedField) == false)) {
+                            int dot = fieldName.lastIndexOf('.');
+                            /*
+                             * Looking up the "tree" at the parent fields here to see if the field is an alias.
```

Comment:
We discussed offline that this isn't a very solid way of checking whether a field is an alias. The fact that we don't return parent information about field aliases is kind of an 'accident' of the field caps logic, and I don't think it was an intentional API decision.

I filed #44298 to propose adding explicit field alias information to the field caps API, which would help make this more robust. It might also give enough information to be able to resolve + extract the alias field values from `_source`.